demos/http3/ossl-nghttp3-demo-server.c

```diff
@@ -0,0 +1,723 @@
+                   &result_count);
+    if (!ret) {
+        fprintf(stderr, "SSL_poll failed\n");
+        return -1; /* something is wrong */
+    }
+    if (result_count == 0) {
+        /* Timeout may be something somewhere */
+        return 0;
+    }
+
+    /* We have something */
+    item = items;
+    for (i = 0; i < numitem; i++) {
+        item++;
+        if (item->revents == SSL_POLL_EVENT_NONE) {
+            continue;
+        } else if (item->revents == SSL_POLL_EVENT_W) {
+            printf("revent %llu (%d) on %llu\n",
+                   (unsigned long long)item->revents, SSL_POLL_EVENT_W,
+                   (unsigned long long)ssl_ids[i].id);
+        } else {
+            /* Figure out ??? */
+            printf("revent %llu (%d) on %llu\n",
+                   (unsigned long long)item->revents, SSL_POLL_EVENT_NONE,
+                   (unsigned long long)ssl_ids[i].id);
+            hassomething++;
+        }
+    }
+    return hassomething;
+}
+
+/* The crappy test wants 20 bytes */
+static uint8_t nulldata[20] = "12345678901234567890";
+static nghttp3_ssize step_read_data(nghttp3_conn *conn, int64_t stream_id,
+                                    nghttp3_vec *vec, size_t veccnt,
+                                    uint32_t *pflags, void *user_data,
+                                    void *stream_user_data) {
+    struct h3ssl *h3ssl = (struct h3ssl *)user_data;
+
+    if (h3ssl->datadone) {
+        *pflags = NGHTTP3_DATA_FLAG_EOF;
+        return 0;
+    }
+    vec[0].base = nulldata;
+    vec[0].len = 20;
+    h3ssl->datadone++;
+
+    return 1;
+}
+
+static int quic_server_write(struct h3ssl *h3ssl, uint64_t streamid,
+                             uint8_t *buff, size_t len, uint64_t flags,
+                             size_t *written) {
+    struct ssl_id *ssl_ids;
+    int i;
+
+    ssl_ids = h3ssl->ssl_ids;
+    for (i = 0; i < MAXSSL_IDS; i++) {
+        if (ssl_ids[i].id == streamid) {
+            if (!SSL_write_ex2(ssl_ids[i].s, buff, len, flags, written) ||
+                *written != len) {
+                fprintf(stderr, "couldn't write on connection\n");
+                ERR_print_errors_fp(stderr);
+                return 0;
+            }
+            printf("written %ld on %lld flags %lld\n", (unsigned long)len,
+                   (unsigned long long)streamid, (unsigned long long)flags);
+            return 1;
+        }
+    }
+    printf("quic_server_write %ld on %lld (NOT FOUND!)\n", (unsigned long)len,
+           (unsigned long long)streamid);
+    return 0;
+}
+
+#define OSSL_NELEM(x) (sizeof(x) / sizeof((x)[0]))
+
+/*
+ * This is a basic demo of QUIC server functionality in which one connection at
+ * a time is accepted in a blocking loop.
+ */
+
+/* ALPN string for TLS handshake. We pretent h3-29 and h3 */
+static const unsigned char alpn_ossltest[] = {5,   'h', '3', '-', '2',
+                                              '9', 2,   'h', '3'};
```

Comment:
Space after { and before '}' as well as before '9'?
```suggestion
static const unsigned char alpn_ossltest[] = { 5,   'h', '3', '-', '2',
                                               '9', 2,   'h', '3' };
```